add-on/panels/js/copy.jsx

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global Components */
+let { utils: Cu } = Components;
+
```

Comment:
If you could try setting up some simple unit tests here stubbing out the chrome things, that'd be great.  That said, I'd suggest timeboxing this so that we can still try and land tomorrow.